### PR TITLE
majestic million feed

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] ~~In Progress - pending design~~ Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/24432

## Description
A feed consuming the most popular 1M Domains, including mapping.
Added Domain indicator fields.
Added support for a limit in the fetch IOC functionality in the CSV API module.

Screen Shot 2020-10-09 at 23 32 49

## Minimum version of Demisto
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/FeedMajesticMillion/pack_metadata.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Majestic Million Feed",
+    "description": "TODO",
```
Comment (yuvalbenshalom) [inline Packs/FeedMajesticMillion/pack_metadata.json:3]: add description

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good.
please add descriptions and readme to the feed.
lets talk about performance testing.

**Comment by bakatzir** [comment]
@kirbles19 please review the readme, detailed_desc and yml of this PR, aside from the RN.
one note - as it is ingesting up to 1M indicators, i've added a troubleshooting message regarding performance both in the readme and the detailed desc, please have a look at it as well.

**Comment by bakatzir** [comment]
I've added more memory to majestic million as there are 1m IOCs.
alienvaultfeed seems to fail from time to time on exceeding the docker memory as well, might be related to the fact that both of them are running on the same machine.

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 2]
```diff
@@ -236,10 +237,13 @@ def create_fields_mapping(raw_json: Dict[str, Any], mapping: Dict[str, Union[Tup
         if not raw_json.get(field):  # type: ignore
             continue
 
-        try:
-            field_value = re.match(regex_extractor, raw_json[field]).group(1)  # type: ignore
-        except Exception:
+        if not regex_extractor:  # do not run the regex match only if a regex_extractor does not exist
```
Comment (yuvalbenshalom) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py:240]: I think you can drop the comment, it is clear without.

File: Packs/FeedMajesticMillion/Integrations/MajesticMillion/MajesticMillion_description.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+## Majestic Million Feed
+Free search and download of the top million websites.
```
Comment (yuvalbenshalom) [inline Packs/FeedMajesticMillion/Integrations/MajesticMillion/MajesticMillion_description.md:2]:
```suggestion
Free search and download of the top million websites.
This feed integration ingests the top most common web sites addresses as 'good' indicators.
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good.
see my comments.

**Review by yuvalbenshalom — APPROVED** [review]

**Comment by bakatzir** [comment]
@ShirleyDenkberg Any more comments? please review the README.md file with regards to the performance issues that may arise when fetching 1M indicators.

**Comment by ShirleyDenkberg** [comment]
@bakatzir Doc review is done.

**Pull request merged**